tools/helios-build/src/main.rs

```
@@ -1956,92 +1918,154 @@ fn cmd_image(ca: &CommandArg) -> Result<()> {
-        "phbl",
-    ])?;
-    let ahibdir = top_path(&["projects", "amd-host-image-builder"])?;
-    let ahibargs_base = {
-        let mut t: Vec<String> = vec![ahib.as_str().into()];
-        for blob in amdblobs {
-            t.push("-B".into());
-            t.push(blob.to_str().unwrap().into());
-        }
-        t
-    };
-    let ahibargs = {
-        let mut t = ahibargs_base.clone();
 
-        t.push("--config".into());
-        t.push(amdconf.to_str().unwrap().into());
-
-        t.push("--output-file".into());
-        t.push(rom.to_str().unwrap().into());
-
-        t.push("--reset-image".into());
-        t.push(reset.to_str().unwrap().into());
-
-        t
-    };
-    ensure::run_in(
-        log,
-        &ahibdir,
-        &ahibargs.iter().map(String::as_str).collect::<Vec<_>>(),
-    )?;
-    tar.add_file(&rom, "rom")?;
-
-    if ddr_testing {
-        /*
-         * The configuration for amd-host-image-builder is stored in JSON5
-         * format.  Read the file as a generic JSON object:
-         */
-        let f = std::fs::read_to_string(&amdconf)?;
-        let inputcfg: serde_json::Value = json5::from_str(&f)?;
-
-        for limit in [1600, 1866, 2133, 2400, 2667, 2933, 3200] {
-            let romname = format!("rom.ddr{limit}");
-            let rom = rel_path(Some(&outdir), &[&romname])?;
+    /*
+     * Go through and create the per-board phase 1 images.
+     */
+    for (board, board_obj) in board_objs {
+        info!(log, "checking board: {}", board);
+
+        let efs = board_obj
+            .get("efs")
+            .and_then(|v| v.as_str())
+            .unwrap_or("<missing efs>");
+        let efsfile = if efs.starts_with("/") {
+            let p = PathBuf::from(efs);
+            assert!(p.is_absolute());
+            p
+        } else {
+            top_path(&["image", "amd", &efs])?
+        };
+        if !efsfile.is_file() {
+            bail!("AMD configuration file {efs:?} does not exist");
+        }
```

We should do these checks earlier, when we read the file, rather than after some of the work is underway.  It would be good to fail fast, basically, if there's something missing or incorrect, rather than after potentially minutes of builds and a bunch of files changed in the file system, etc.